### PR TITLE
[PC-6250] Add page header for SearchFilter

### PR DESCRIPTION
**Pull request by justinemignot**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-6250

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [x] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.

## Screenshots

| Description | Search | 
| -------------------- | :----------------------: | 
|                 image    |        image       |        

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn version:testing` (this will create a commit with a tag)
- then run `git push --follow-tags`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -220,3 +220,12 @@ To do so:
...
 4. You can merge :)
+
+## Test data
+
+### Accounts
+
+You can use test accounts:
+
+- email: `pctest.jeune93.has-booked-some@btmx.fr`
+- mdp: `user@AZERTY123`
```
Comment (blixit) [inline README.md:231]: 👍

File: src/features/search/pages/SearchFilter.tsx [hunk 2]
```diff
@@ -1,16 +1,37 @@
...
 
-export const SearchFilter: React.FC = () => (
-  <Container>
-    <Spacer.Flex />
-    <Typo.Hero>{_(t`SearchFilter`)}</Typo.Hero>
-    <Spacer.Flex />
-  </Container>
-)
+export const SearchFilter: React.FC = () => {
+  const [rightComponentWidth, setRightComponentWidth] = useState(0)
```
Comment (antoinewg) [inline src/features/search/pages/SearchFilter.tsx:11]: je verrais bien cette logique dans PageHeader (si tu wrappes rightComponent avec ` <View onLayout={onRightComponentLayout}>{rightComponent}</View>` ) pour éviter de le répéter à chaque fois. Tu peux même faire un hook useElementWidth ?
const {width, onLayout} = useElementWidth()

```js
const useElementWidth = () => {
  const [elementWidth, setElementWidth] = useState<number>(0)

  const onLayout = (event: any) => {
    if (elementWidth !== 0) return
    const { width } = event.nativeEvent.layout
    setElementWidth(width)
  }

  return { width: elementWidth, onLayout }
}
```

**Review by antoinewg — COMMENTED** [review]
2 suggestions, sinon ça me parait bien

File: src/ui/components/headers/PageHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  rightComponent?: Element
+  rightComponentWidth?: number
+}
+
+export const PageHeader: React.FC<Props> = (props) => {
+  const { title, rightComponent, rightComponentWidth } = props
+  const { goBack } = useNavigation()
+  const spaceToAddBeforeTitle = rightComponentWidth ? rightComponentWidth / 6 : 0
+
+  const HeaderIconBack = () => {
```
[the remark this answers is not in the document]
Comment (justinemignot) [reply src/ui/components/headers/PageHeader.tsx:20]: done :)

File: src/features/search/pages/SearchFilter.tsx (hunk 2 above)
Comment (justinemignot) [reply src/features/search/pages/SearchFilter.tsx]: 👍

**Review by antoinewg — APPROVED** [review]

**Pull request merged**